docs/proposal/2022-09-28-event-manager-extensibility.md

```diff
@@ -0,0 +1,302 @@
+it can resolve plugin by its prefix which is `kubectl` and call `KubectlExecutor` gRPC endpoint automatically and parses the response.
+As you can also see, Botkube does not need to know how `kubectl` works, it simply delegates all the responsibility to specific plugin.
+
+![](./assets/kubectl-executor-plugin.png)
+
+### Cloud Events
+Cloudevents schema has a proven standard for the cloud native industry where well-known products uses this schema to distribute their events
+via this contract. The main take of this, you don't need to re-invent wheel from scratch while you are designing your data format. In Botkube,
+we can also use [one of the Cloudevents SDK](https://github.com/cloudevents/sdk-go) to spin up an handler inside Botkube so that any cloudevents 
+compatible source can send data to Botkube. The main example to this is [Keptn](https://keptn.sh/), an open source application life-cycle orchestration tool, 
+they send events in cloudevents format. If we want to ship Keptn events to Botkube to be able to send notifications, we may need to add new integration to Keptn
+so that it will send events to Botkube `/events` endpoint which handles cloudevents.
+
+#### Botkube Cloudevents Data Structure
+You will see very simple suggested event format for Botkube as shown below.
+```json
+{
+  "id" : "e2361318-e50b-472e-8b17-13dbac1daac1",   
+  "source" : "Kubernetes",     
+  "specversion" : "1.0",                             
+  "time" : "2022-03-23T14:35:39.738Z",               
+  "data" : {                                         
+    "namespace" : "Botkube",
+    "deployment" : {
+      "name" : "test",
+      "replicas" : 5
+    }
+  }
+}
+```
+
+`id`: This is the unique identifier for the event, and this should be decided by end users. UUID is not a mandatory stuff, it can
+be also ARN for example `ec2:myinstance:12345`.
+
+`source`: This refers to plugin name. Once Botkube receives and event on `/events` endpoint, `source` field would be validated
+to see we already have a registered plugin for this event or not. 
+
+`specversion`: To have proper versioning for Botkube event format.
+
+`time`: Event time
+
+`data`: This is the event data that contains important information about source. what you see in `data` section is just example,
+we can also add some kind of validation in Botkube to better use received data. For example, if we force user to send key value pairs,
+we can easily apply templating/filtering to this event to full-fill our future stories about sending custom notifications to slack.
+Data structure is strongly opened to suggestion.
+
+As you can also understand, data field contains generic data which can vary for plugins. For internal usage, we can create a general event data structure, but
+I haven't included the final structure for that, I believe it can be generated from scratch during real implementation.
```

Comment:
Let's state that in the consequences as this will also impact the estimation

Reply:
Stated in the consequences